pkg/authhelpers/gcpauth_test.go

```diff
@@ -0,0 +1,87 @@
+package authhelpers
+
+import (
+	"context"
+	"encoding/json"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+//nolint:paralleltest
+func TestResolveGoogleCredentials_ValidCredentials(t *testing.T) {
+	t.Setenv("GOOGLE_CREDENTIALS", `{
+		"type": "service_account",
+		"project_id": "your-project-id",
+		"private_key_id": "your-private-key-id",
+		"private_key": "your-private-key",
+		"client_email": "your-client-email",
+		"client_id": "your-client-id"
+	}`)
+
+	ctx := context.Background()
+	scope := "some-scope"
+
+	credentials, err := ResolveGoogleCredentials(ctx, scope)
+
+	assert.NoError(t, err)
+	assert.NotNil(t, credentials)
+
+	var creds map[string]interface{}
+	err = json.Unmarshal([]byte(os.Getenv("GOOGLE_CREDENTIALS")), &creds)
+	assert.NoError(t, err)
+	assert.Equal(t, creds["type"], "service_account")
+	assert.Equal(t, creds["project_id"], "your-project-id")
+	assert.Equal(t, creds["private_key_id"], "your-private-key-id")
+	assert.Equal(t, creds["private_key"], "your-private-key")
+	assert.Equal(t, creds["client_email"], "your-client-email")
+	assert.Equal(t, creds["client_id"], "your-client-id")
+
+	t.Cleanup(func() {
```

Comment:
The `t.Setenv` should handle this, no need for an explicit cleanup.

Reply:
good point thanks!